modules/config.py

```diff
@@ -121,9 +123,13 @@ def _config(self, export_settings, target, setting, value=None):
                 raise ConfigInvalidValue(str(e))
 
             if not validated_value == None:
-                target.set_setting(setting, validated_value)
+                existing_value = target.get_setting(setting, None)
                 formatted_value = setting_object.format(validated_value)
```

Comment:
can this line also be moved to after we've decided the value has changed?

Reply:
yes it can, good pickup